src/command.js

```diff
@@ -221,4 +228,31 @@ export default class StorybookCommand extends Command {
       await page?.close();
     }
   }
+
+  async getStorybookVersion(url) {
+    let version, page;
+
+    try {
+      page = await this.percy.browser.page();
+      this.log.debug(`Get Storybook version: ${url}/?path=/settings/about`);
+
+      await page.goto(`${url}/?path=/settings/about`);
+      /* istanbul ignore next: no instrumenting injected code */
+      version = await page.eval(async ({ waitFor }) => {
+        await waitFor(() => !!document.querySelector('header'), 5000)
+          .catch(() => Promise.reject(new Error('Failed to find a <header> element')));
+        let getPath = path => document.evaluate(path, document, null, 9, null).singleNodeValue;
+        let text = getPath("//header[starts-with(text(), 'Storybook ')]").innerText;
+
+        return text.match(/[-]{0,1}[\d]*[.]{0,1}[\d]+/g, '').join('');
+      });
+    } catch (error) {
+      this.log.debug(`Couldn't retrieve Storybook version: ${error.message}`);
```

Comment:
In the matching eval pattern for story collection, we remove ugly stack traces from any error messages. We can move that regexp into a constant, and/or make a small helper function to handle eval errors.